sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -241,27 +246,170 @@ private object CosmosSchemaInferenceConfig {
+  private[spark] val Now = Value("Now")
+  private[spark] val PointInTime = Value("PointInTime")
+}
+
+private case class CosmosChangeFeedConfig
+(
+  changeFeedMode: ChangeFeedMode,
+  startFrom: ChangeFeedStartFromMode,
+  startFromPointInTime: Option[Instant],
+  maxItemCountPerTrigger: Option[Long]
+)
 
 private object CosmosChangeFeedConfig {
-  private val DefaultChangeFeedMode: ChangeFeedMode = ChangeFeedModes.incremental
+  private val DefaultChangeFeedMode: ChangeFeedMode = ChangeFeedModes.Incremental
+  private val DefaultStartFromMode: ChangeFeedStartFromMode = ChangeFeedStartFromModes.Beginning
+
+  private[spark] val startFrom = CosmosConfigEntry[ChangeFeedStartFromMode](
+    key = "spark.cosmos.changeFeed.startFrom",
+    mandatory = false,
+    parseFromStringFunction = startFromNotYetValidated => this.validateStartFromMode(startFromNotYetValidated),
+    helpMessage = "ChangeFeed Start from settings (Now, Beginning  or a certain point in " +
+      "time (UTC) for example 2020-02-10T14:15:03) - the default value is 'Beginning'.")
 
-  private[spark] val changeFeedMode = CosmosConfigEntry[ChangeFeedMode](key = "spark.cosmos.changeFeed.mode",
+  private[spark] val startFromPointInTime = CosmosConfigEntry[Instant](
+    key = "spark.cosmos.changeFeed.startFrom",
+    mandatory = true,
+    parseFromStringFunction = startFrom => Instant.from(DateTimeFormatter
+      .ISO_INSTANT
+      .parse(startFrom.trim)),
+    helpMessage = "ChangeFeed Start from settings (Now, Beginning  or a certain point in " +
+      "time (UTC) for example 2020-02-10T14:15:03Z) - the default value is 'Beginning'.")
+
+  private[spark] val changeFeedMode = CosmosConfigEntry[ChangeFeedMode](
+    key = "spark.cosmos.changeFeed.mode",
     mandatory = false,
-    parseFromStringFunction = changeFeedModeString => ChangeFeedModes.withName(changeFeedModeString),
+    parseFromStringFunction = changeFeedModeString => validateChangeFeedMode(changeFeedModeString),
     helpMessage = "ChangeFeed mode (Incremental or FullFidelity)")
 
+  private[spark] val maxItemCountPerTriggerHint = CosmosConfigEntry[Long](
+    key = "spark.cosmos.changeFeed.maxItemCountPerTriggerHint",
+    mandatory = false,
+    parseFromStringFunction = maxItemCount => maxItemCount.toInt,
+    helpMessage = "Approximate maximum number of items read from change feed for each trigger")
+
+  private def validateChangeFeedMode(mode: String): ChangeFeedMode = {
+    Option(mode).fold(DefaultChangeFeedMode)(p => {
+      val modeName = p.trim
+
+      if (modeName.isEmpty) {
+        DefaultChangeFeedMode
+      } else {
+        ChangeFeedModes
+          .values
+          .find(_.toString.equalsIgnoreCase(modeName))
+          .getOrElse(throw new IllegalArgumentException(s"Invalid change feed mode '$modeName'"))
+      }
+    })
+  }
+
+  private def validateStartFromMode(startFrom: String): ChangeFeedStartFromMode = {
+    Option(startFrom).fold(DefaultStartFromMode)(sf => {
```

Comment:
this parses a string from enum. We have multiple cases of parsing string to enum (for different enums).

we should later consider writing a common code for enum parsing and re-use everywhere. This is outside of the scope of the PR just a TODO for later.

Reply:
Agreed - added comment.